notebooks/build_index_alternative.R

```diff
@@ -0,0 +1,181 @@
+# ---- Setup ----
+library(tidyverse)
+library(psych)
+library(ggcorrplot)
+library(here)
+
+indir <- here("data", "clean", "england", "ltla")
+
+# --- Load, clean and merge datasets ----
+# Note: Code to clean and merge datasets largely from build_index.R
+
+# --- Load datasets ----
+files <- list.files(indir, pattern = "eng_.*.csv", full.names = TRUE)
+dat <- Map(read.csv, files)
+
+names(dat) <- basename(names(dat))
+
+# 1) duplicated dataset entries:
+dat <- dat[names(dat) != "eng_asylum_support_ltla.csv"]
+
+# 2) any datasets do not have columns starting with "z_" ?
+which(!mapply(any, Map(grepl, Map(colnames, dat), pattern = "z_")))
+
+# 3) any datasets without "ltla24_code" column ?
+which(!mapply(any, Map(`==`, Map(colnames, dat), "ltla24_code")))
+
+# one dataset had a column named "ltla_cd" instead if "ltla_code"
+# two other datasets had columns named "lad_code" instead of "ltla_code"
+# otherwise the entries matched with ltla24 codes, so assuming mistakes in naming
+colnames(dat[["eng_care_home_residents_ltla24.csv"]])[2] <- "ltla24_code"
+colnames(dat[["eng_asylum_support_ltla24.csv"]])[2] <- "ltla24_code" # NOTE: this data also had fewer rows
+# colnames(dat[["eng_homeless_hh_ltla.csv"]])[2]    <- "ltla24_code"
+
+# --- Merge datasets ----
+# only select ltla24 codes and z score data from each dataset
+dat <- Map(function(x) x[, grepl("ltla24_|z_", colnames(x))], dat)
+
+# merge all the datasets by ltla24_code column
+dat <- Reduce(function(x, y) merge(x, y, by = "ltla24_code", all.x = TRUE), dat)
+
+# z_asylum_support_perc had fewer rows imputing the missing values with 0
+# note that for z-scores this is equivalent with mean imputation
+dat[is.na(dat)] <- 0
+
+# check if all columns are proper z-scores
+# they should have mean equal to 0 and sd equal to 1
+which(mapply(mean, dat[-1]) > 0.01)
+which(mapply(sd, dat[-1]) > 1.01)
+
+# FIXME: looks like z_english_poor_none_perc is not a proper z-score
```

Comment:
Did you plot the distribution prior to transformation? What did it look like? Curious as to why this wasn't a proper z-score and if there is a reason for it we need to factor in.

Reply:
@jennajt this is for you.